gcputil/credentials.go

```diff
@@ -146,23 +156,46 @@ func PublicKey(pemString string) (interface{}, error) {
 	return cert.PublicKey, nil
 }
 
-func ServiceAccountPublicKey(serviceAccount string, keyId string) (interface{}, error) {
-	keyUrl := fmt.Sprintf(serviceAccountPublicKeyUrlTemplate, url.PathEscape(serviceAccount))
-	res, err := cleanhttp.DefaultClient().Get(keyUrl)
+// ServiceAccountPublicKey returns the public key with the given key ID for
+// the given service account if it exists. If the key does not exist, an error
+// is returned.
+func ServiceAccountPublicKey(ctx context.Context, serviceAccount, keyID string) (interface{}, error) {
```

Comment:
Just realized that adding context here breaks the exported API. I wasn't able to find anything else using this besides Vault GCP auth, but best to not break it. I will add a function with the same signature from before that doesn't take a context.

Reply:
Okay, exported API should be unchanged with c63b976.